prow/config/jobs.go

```diff
@@ -17,12 +17,62 @@ limitations under the License.
 package config
 
 import (
+	"fmt"
 	"regexp"
 	"time"
 
 	"k8s.io/test-infra/prow/kube"
 )
 
+// Preset is intended to match the k8s' PodPreset feature, and may be removed
```

Comment:
I don't think we want to hide job configuration in the cluster config, it should be possible to load this config and know exactly what the pod looks like in a unit test so we can validate configurations and so we don't have to worry about what version of the preset exists on the cluster.

Reply:
Once it goes to beta you shouldn't have to worry about the version since there is a N+2 deprecation policy and existing features of the preset shouldn't break.

Reply:
I'm not worried about the version, but I don't want to hide pod definitions outside of the prow config, and using *actual* pod presets moves configuration state into the API server. 

We need to be able to validate pod configurations.

Reply:
You will likely store pod presets the same way you store pod specs. Also, that's what this PR is doing.

Reply:
Sure, but it mentions removing it... Since we don't consume the upstream types that would imply to me using them natively in kubernetes instead of loading them in the config agent. 

We should definitely continue to handle them in prow so we can process the fully loaded form reliably.

Reply:
> Sure, but it mentions removing it...

I think this is about removing the custom structure @spxtr introduces here but not removing the field from `Config`, ie. reuse of the actual API from k8s. Maybe I misuderstood.

Reply:
Yes, that's right.

Reply:
Just realized I never replied to this thread, SGTM. Sorry I misunderstood here 😬